strapi/strapi
# [DEITS] Github workflow to compile monorepo typescript packages

### What does it do?

Updates the github workflow to compile typescript before running lint and unit tests

"Compile" is proposed as the terminology specifically for compiling code, whereas "build" within the Strapi monorepo refers to processes such as building the front-end files, webpack, helper plugin creating an index file, etc.

It's necessary to separate the two, because "compile" (at least for now) is a very fast process, whereas "build" takes a considerable amount of time that would slow down the workflow.

### Why is it needed?

1. The linting rules `import/no-unresolved` and `node/no-missing-require` fail when including Typescript packages (ie, @strapi/data-transfer) because the files do not exist
2. It fixes a circular lerna dependency we have (strapi/admin:build -> strapi/data-transfer:build -> strapi/admin:build) but I don't totally understand why, since this is doing the exact same thing with different words.
3. Javascript code couldn't import typescript packages like @strapi/data-transfer because the CI checks would fail when the compiled package wasn't found

### How to test it?

Github checks should all succeed even though we now access our typescript packages from javascript.

It shouldn't add a considerable delay. Locally, it takes me about 2 seconds to run. Although that will increase as Strapi adds more Typescript to the codebase, it is a necessary step if we plan to move in the direction of Typescript.

### Related issue(s)/PR(s)

#15078
